app/scripts/migrations/120.2.test.ts

```diff
@@ -265,6 +306,61 @@ describe('migration #120.2', () => {
       expect(transformedState.data).toEqual(oldState);
     });
 
+    it('does nothing if obsolete properties are not set and providerConfig is set to undefined', async () => {
+      const oldState = {
+        NetworkController: {
+          // This should be impossible because `undefined` cannot be returned from persisted state,
```

Comment:
I discovered this when investigating this test failure: https://github.com/MetaMask/metamask-extension/actions/runs/10216054639/job/28266703200